src/lib/executors/Node.ts

```diff
@@ -875,13 +876,34 @@ export default class Node extends Executor<NodeEvents, Config, NodePlugins> {
         }
 
         testTask
-          .then(() => {
+          .then(async () => {
             if (!this._sessionSuites) {
               return;
             }
-            return this._loadFunctionalSuites().then(
-              () => (testTask = this._runRemoteTests())
-            );
+
+            let remainingAttempts = 1 + (this.config.functionalRetries || 0);
+            let suites = this._sessionSuites;
+
+            await this._loadFunctionalSuites();
+            while (remainingAttempts && suites.length) {
+              remainingAttempts--;
+              if (suites.length !== this._sessionSuites!.length) {
+                this.log(
+                  'reattempting',
+                  suites.length,
+                  'of',
+                  this._sessionSuites!.length,
+                  'environments'
+                );
+              }
+              testTask = this._runRemoteTests(suites);
+              await testTask.catch(remainingAttempts ? () => {} : undefined);
+              suites = suites.filter(suite => suite.failed);
```

Comment:
Consider using a meaningful name here (maybe `failingSuites`), just to make it clearer what's being done in the following if statement, then assign that to `suites` at the end of the loop.

Reply:
Hmm... The conditional check is on the next line. Do you think it needs an explicit name when usage is so close to assignment?

Reply:
It makes it read better. It was the usage specifically that prompted the comment; it's not immediately apparent when skimming through the code that `if (suites.length === this._sessionSuites.length)` is comparing a list of failing suites to the original list.